neon_mq_connector/utils/client_utils.py

```diff
@@ -94,22 +100,31 @@ def handle_mq_response(channel: Channel, method, _, body):
         """
         api_output = b64_to_dict(body)
 
-        # The Messagebus connector generates a unique `message_id` for each
-        # response message. Check context for the original one; otherwise,
-        # check in output directly as some APIs emit responses without a unique
-        # message_id
+        # Backwards-compat. handles `context` in response for raw `Message`
+        # objects sent across the MQ bus
         api_output_msg_id = \
             api_output.get('context',
                            api_output).get('mq', api_output).get('message_id')
-        # TODO: One of these specs should be deprecated
         if api_output_msg_id != api_output.get('message_id'):
-            LOG.debug(f"Handling message_id from response context")
+            # TODO: `context.mq` handling should be deprecated
+            LOG.warning(f"Handling message_id from response context")
         if api_output_msg_id == message_id:
             LOG.debug(f'MQ output: {api_output}')
             channel.basic_ack(delivery_tag=method.delivery_tag)
-            channel.close()
-            response_data.update(api_output)
-            response_event.set()
+            if isinstance(api_output.get('_part'), int):
+                # TODO: Consider forcing these to be passed to `stream_callback` synchronously
+                # Handle multi-part responses
+                if stream_callback:
+                    # Pass each part to the stream callback method if defined
+                    stream_callback(api_output)
+                if api_output.get('_is_final'):
+                    # Always return final result
+                    response_data.update(api_output)
+            else:
+                response_data.update(api_output)
```

Comment:
Does it mean that the only way to send objects is not chunk by chunk but only solid object but with partial additions to it's content?

Reply:
That is the intended use case as [documented in the README changes](https://github.com/NeonGeckoCom/neon_mq_connector/pull/101/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R72-R77).

I implemented the stream callback as optional, so `send_mq_request` is expected to always return a complete result at the end.

Chunked responses would require specifying a standard method for combining them into a single complete response which is also valid, but would require some additional information in responses, i.e.:
- how to combine response data
- how to present incremental responses to a consumer